BUILD

```diff
@@ -2915,17 +2915,22 @@ grpc_cc_library(
         "src/core/lib/event_engine/slice_buffer.cc",
     ],
     hdrs = [
+        "include/grpc/event_engine/slice.h",
+        "include/grpc/event_engine/slice_buffer.h",
```

Comment:
Why don't depend on `event_engine_base_hdrs`?

Reply:
hdrs specifies that this target exports these headers (and in this case provides an implementation) - event_engine_base_hdrs (and the equivalent only-headers targets) largely should be removed, I just haven't got there yet.

Reply:
I feel worried about removing all only-headers targets. They are magical keys to our build file to avoid cyclic dependencies in many occasions. Removing `event_engine_base_hdrs` itself should be ok.

Reply:
Right... we should at some point in the future figure out why magic is required and make it not required.

In the meantime - the implementation of Slice, SliceBuffer that's advertised from these includes is supplied by this library, so this library should export those headers.

Reply:
It would be a huge refactoring I imagine.

Reply:
It's important - for the dependency analysis to work we need to be able to tie the headers and the implementations into bazel targets. The public headers without the backing implementation pattern we have is an anti-pattern and it eventually needs to go - the only "header only" targets we should have are the ones which have no backing cc files. Our destination state should be: if the build system allows me to include a header file I should be able to call all functions declared within that file. That is not true now, and sometimes in surprising ways.

In the meantime, the following targets adopt this (correct) pattern: `grpc++_binder`, `grpc++_xds_server`, `grpc++_error_details`, `grpc++_alts`, `census`, `gpr_atm`, `gpr_platform`, `event_engine_memory_allocator`, `slice_refcount`, `slice`, `slice_buffer`, `grpcpp_server_load_reporting`, `grpc++_authorization_provider`, `grpc++_reflection`, `grpcpp_call_metric_recorder`, `grpcpp_orca_service`, `grpcpp_channelz`, `grpcpp_admin`, `grpc++_test`, `grpc_opencensus_plugin`.

Reply:
That sounds promising. The only thing I feel worried about now is those targets needing `select_deps`. I think we can divide targets properly to avoid build issues. We need to be careful.